src/views/exportJarFile.ts

```diff
@@ -0,0 +1,335 @@
+            } else {
+                reject();
+            }
+        });
+    }
+
+    private static resolveMainMethod(progress, token: CancellationToken, pickSteps: string[], projectPath: string): Promise<string> {
+        return new Promise<string>(async (resolve, reject) => {
+            if (token.isCancellationRequested) {
+                return reject();
+            }
+            progress.report({ increment: 10, message: "Resolving main classes..." });
+            if (this.mainMethods === undefined || this.mainMethods.length === 0) {
+                return resolve("");
+            }
+            progress.report({ increment: 30, message: "Determining entry main class..." });
+            const pickNodes: QuickPickNode[] = [];
+            for (const mainMethod of this.mainMethods) {
+                if (Uri.file(mainMethod.path).fsPath.includes(projectPath)) {
+                    pickNodes.push(new QuickPickNode(this.getName(mainMethod), mainMethod.name));
+                }
+            }
+            if (pickNodes.length === 0) {
+                return resolve("");
+            } else {
+                const pickBox = window.createQuickPick<QuickPickNode>();
+                pickNodes.push(new QuickPickNode("No main class", ""));
+                pickBox.items = pickNodes;
+                pickBox.title = "Export Jar - Determine entry main class";
+                pickBox.placeholder = "Select the entry main class...";
+                pickBox.ignoreFocusOut = true;
+                pickBox.buttons = pickSteps.length > 0 ? [(QuickInputButtons.Back)] : [];
+                pickBox.onDidTriggerButton((item) => {
+                    if (item === QuickInputButtons.Back) {
+                        reject(InputFlowAction.back);
+                        pickBox.dispose();
+                    }
+                });
+                pickBox.onDidAccept(() => {
+                    pickSteps.push(SOLVE_MAINMETHOD);
+                    resolve(pickBox.selectedItems[0].description);
+                    pickBox.dispose();
+                });
+                pickBox.onDidHide(() => {
+                    reject();
+                    pickBox.dispose();
+                });
+                pickBox.show();
+            }
+        });
+    }
+
+    private static failMessage(message: string) {
+        window.showInformationMessage<Message>(message, new Option(false), new Message("Done", true));
+    }
+
+    private static successMessage(outputFileName: string) {
+        let openInExplorer: Message;
+        if (platform() === "win32") {
+            openInExplorer = new Message("Reveal in File Explorer");
+        } else if (platform() === "darwin") {
+            openInExplorer = new Message("Reveal in Finder");
+        } else {
+            openInExplorer = new Message("Open Containing Folder");
+        }
+        window.showInformationMessage<Message>("Successfully exported jar to" + EOL + outputFileName,
```

Comment:
If you do not want to have a modal dialog, please directly use `export function showInformationMessage(message: string, ...items: string[]): Thenable<string | undefined>;`